### DISCUSSION TITLE
Close PRs for libraries that update constantly

### DISCUSSION THREAD

**Discussion by rcdailey** [renovatebot/renovate#32276] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I have a PR for a nuget packaged called MudBlazor here: https://github.com/recyclarr/recyclarr/pull/347

This library has a lot of churn; there are release versions constantly. For whatever reason, the PR got open despite my `minimumReleaseAge` being set to 5 (Maybe 5 days really did pass and I just missed it, but that's not what's important).

Now that the PR is already open, I typically neglect merging it until I see there are no pending versions. I want to see that it is stable. I've also customized its min release age to 10 to help with this:

```json
{
  "matchManagers": ["nuget"],
  "matchPackageNames": ["MudBlazor"],
  "minimumReleaseAge": "10 days",
}
```

Is there a way to tell Renovate to close the PR for this package if there are pending versions? I really don't even want to see it in my PR list unless it's been stable for 10 days *and* has no pending versions.

### Logs (if relevant)

https://developer.mend.io/github/recyclarr/recyclarr/-/job/d6b1b9f6-ea5a-4151-9218-726f2f255c33

**Comment by rarkins** [comment, marked as answer]
Close the PR and ensure that the branch is deleted. Hopefully on the run after that the branch is not created

**Reply by rcdailey** [reply]
I thought closing a PR resulted in Renovate not reopening again under different conditions, such as next major version? I probably have this wrong.

**Reply by rarkins** [reply]
Only for that exact version in the PR

**Reply by rcdailey** [reply]
Looks like it popped up again... https://github.com/recyclarr/recyclarr/pull/358

`internalChecksFilter` docs say the default is `strict` so I would expect the PR to not be opened until the minimum release age is met or exceeded.

**Reply by rarkins** [reply]
Please copy/paste the debug logs for that branch from the run when the branch was created.

**Reply by rcdailey** [reply]
Here are the logs. I can see that the branch gets created in them.

https://developer.mend.io/github/recyclarr/recyclarr/-/job/0192e987-7287-72aa-a720-91852acbc33b

Looks like creating the PR was the correct action for Renovate to take based on this in the logs:

```
                "releaseTimestamp": "2024-10-22T21:12:41.137Z",
                "newVersionAgeInDays": 10,
```

I did set my min age to 10, but I did not imagine it had already been 10 days since that release.

I think the missing behavior here is that this PR has a pending release for 7.15 and I do not want PRs opened unless the min age is met *and* there are no pending releases. In addition, each time there is a pending release, it should reset the clock: The min release age should start from that pending release.

The goal here, is to avoid PR churn for libraries that release very frequently. I am generally not interested in upgrading when libraries get spurts of releases very quickly. This usually, in my experience, indicates some instability and the follow up releases are attempting to resolve issues.

I like the release pattern / cycle for a library to calm down before I ever see a PR opened, since I'd rather not be disturbed by even seeing it until that point.

I'm not sure if this is a flexibility that Renovate offers, but hopefully you can share more on that.

Thank you for your continued support.

**Reply by rarkins** [reply]
That's not what I asked for
